llvm/lib/Transforms/Scalar/StructurizeCFG.cpp

```diff
@@ -990,10 +996,10 @@ BasicBlock *StructurizeCFG::getNextFlow(BasicBlock *Dominator) {
                                         Func, Insert);
   FlowSet.insert(Flow);
 
-  // use a temporary variable to avoid a use-after-free if the map's storage is
-  // reallocated
-  DebugLoc DL = TermDL[Dominator];
-  TermDL[Flow] = std::move(DL);
+  auto *Term = Dominator->getTerminator();
+  if (const DebugLoc &DL =
+          Term ? Term->getDebugLoc() : TermDL.lookup(Dominator))
+    TermDL[Flow] = DL;
```

Comment:
This is a double map lookup in the !Term case

Reply:
First is for the Dominator and the second is for the Flow blocks